persistent/test/Database/Persist/TH/KindEntitiesSpec.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+-- {-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+-- {-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedLabels #-}
+-- {-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TemplateHaskell #-}
+-- {-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Database.Persist.TH.KindEntitiesSpec where
+
+import Database.Persist.TH.KindEntitiesSpecImports
+import TemplateTestImports
+
+mkPersist sqlSettings [persistLowerCase|
+
+Customer
+    name    String
+    age     Int
+
+CustomerTransfer
+    customerId CustomerId
+    moneyAmount (MoneyAmount 'CustomerOwned 'Debit)
+|]
+
+spec :: Spec
+spec = describe "KindEntities" $ do
+    it "works" $ do
+        {-
+        let UserName = #name
+            OrganizationName = #name
+            DogName = #name
+-}
```

Comment:
There might be some more stuff to do here for this test maybe ... for now I have something that just asserts that it typechecks :+1:

Reply:
Yeah - let's check the type of the field accessors and the `EntityField`s.

Still should jsut be a compilation check, but something like:

```haskell
let mkTransfer :: CustomerId -> MoneyAmount 'CustomerOwned 'Debt -> CustomerTransfer
    mkTransfer = CustomerTransfer
    getAmount :: CustomerTransfer -> MoneyAmount 'CustomerOwned 'Debt
    getAmount = customerTransferMoneyAmount
```